test/foundry/unit/TokenDistributorGasCalculation.t.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.19;
+
+import {ITokenDistributor} from "../../../contracts/interfaces/ITokenDistributor.sol";
+import {TokenDistributor} from "../../../contracts/TokenDistributor.sol";
+import {TokenDistributorInternals} from "../utils/TokenDistributorInternals.sol";
+import {Kwenta} from "../../../contracts/Kwenta.sol";
+import {StakingRewardsV2} from "../../../contracts/StakingRewardsV2.sol";
+import {RewardEscrowV2} from "../../../contracts/RewardEscrowV2.sol";
+import {StakingSetup} from "../utils/setup/StakingSetup.t.sol";
+
+
+/// @notice test how many weeks we can go without checkpointing
+/// and how much gas will it cost
+contract TokenDistributorGasCalculation is StakingSetup {
+
+    TokenDistributor public tokenDistributor;
+
+    function setUp() public override {
+        /// @dev starts after a week so the startTime is != 0
+        goForward(1 weeks + 1);
+        super.setUp();
+        switchToStakingV2();
```

Comment:
Do you need `switchToStakingV2()` for any of your tests? My understanding is that your function redirects inflation schedule to StakingV2 but that's only useful if you're minting new rewards.

Reply:
Doesn't all of TokenDistributor rely on StakingV2 because of the V2 checkpointing? Or you mean why did I choose to implement it in my tests by doing switchToStakingV2?